[Hardware][TPU] Multi-LoRA implementation for the TPU backend

Akshat-Tripathi opened the pull request:
This PR adds a Multi-LoRA implementation that works on the TPU backend, extending the work done in #11100.

Currently this uses pytorch operations for the Punica kernels, but I am going to put up a PR with Pallas kernels soon.

liangfu commented on vllm/lora/layers.py:
```diff
@@ -1077,12 +1077,21 @@ def _get_logits(
         lora_logits = lora_logits.mT
         indices_padded = self.punica_wrapper.sampler_indices_padded
+
+        if current_platform.is_tpu():
+            # Because nan_to_num_ doesn't work with actual -inf values on TPU
+            neg_inf = torch.finfo(lora_logits.dtype).min
+            pos_inf = torch.finfo(lora_logits.dtype).max
+        else:
+            neg_inf = float("-inf")
+            pos_inf = float("inf")
```
file an issue with torch-xla ?

jeejeelee replied:
`abstract this as part of an utility function ` sounds good

liangfu replied:
abstract it away as a short-term solution is fine.

it would better if we can create an issue in torch-xla repo, as a longer-term solution.

Akshat-Tripathi replied:
Ok, I've made the issue here: https://github.com/pytorch/xla/issues/8674

jeejeelee commented on vllm/lora/ops/xla_ops/lora_ops.py:
```diff
@@ -0,0 +1,58 @@
+import torch
+
+from ..torch_ops import bgmv_expand, bgmv_expand_slice, bgmv_shrink
```
It seems the TPU ops are still using PyTorch operators,  is it necessary to add the below ops?

Akshat-Tripathi replied:
The `sgmv` ops are slightly different here because I'm using `repeat_interleave` with a static size rather than a dynamic tensor, which reduces the compile time quite a bit because torch_xla can't lower the dynamic version properly.

jeejeelee commented on vllm/lora/punica_wrapper/punica_tpu.py:
```diff
@@ -0,0 +1,330 @@
+
+# The platforms that are compatible with the PyTorch-native implementation can
+# inherit this class
+class PunicaWrapperTPU(PunicaWrapperBase):
```
Why not directly inherit from `PunicaWrapperCPU `?

Akshat-Tripathi replied:
I thought about it, but this code is going to change very soon as I add in the Pallas kernels

Akshat-Tripathi commented:
It looks like the `Async Engine, Inputs, Utils, Worker Test` is failing on multimodal inputs, which is WIP right now.
The TPU test seems to be failing on non lora code. Do these tests pass on main? I'm wondering if they're linked to this PR or something else

miladm commented:
cc @lsy323 to take a pass

WoosukKwon committed:
[V1] Get input tokens from scheduler (#13339)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

comaniac committed:
[V1][PP] Fix intermediate tensor values (#13417)

Signed-off-by: Cody Yu <hao.yu.cody@gmail.com>

WoosukKwon committed:
[V1][Spec decode] Move drafter to model runner  (#13363)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

tlrmchlsmth committed:
[Bugfix][CI][V1] Work around V1 + CUDA Graph + torch._scaled_mm fallback issue (#13425)

Signed-off-by: Tyler Michael Smith <tyler@neuralmagic.com>

hmellor committed:
[Misc] Remove dangling references to `SamplingType.BEAM` (#13402)

Isotr0py committed:
[Model] Enable quantization support for `transformers` backend (#12960)

divakar-amd committed:
[ROCm] fix get_device_name for rocm (#13438)

Signed-off-by: Divakar Verma <divakar.verma@amd.com>

youkaichao committed:
[v1] fix parallel config rank (#13445)

Signed-off-by: youkaichao <youkaichao@gmail.com>

kylesayrs committed:
[Quant] Molmo SupportsQuant (#13336)

kylesayrs committed:
[Quant] Arctic SupportsQuant (#13366)

terrytangyuan committed:
[Bugfix] Only print out chat template when supplied (#13444)

youkaichao committed:
[core] fix sleep mode in pytorch 2.6 (#13456)

Signed-off-by: youkaichao <youkaichao@gmail.com>

kylesayrs committed:
[Quant] Aria SupportsQuant (#13416)

WoosukKwon committed:
[V1][PP] Fix & Pin Ray version in requirements-cuda.txt (#13436)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

mgoin committed:
Add outlines fallback when JSON schema has enum (#13449)

Signed-off-by: mgoin <mgoin64@gmail.com>

terrytangyuan committed:
[Bugfix] Ensure LoRA path from the request can be included in err msg (#13450)

Signed-off-by: Yuan Tang <terrytangyuan@gmail.com>

Isotr0py committed:
[Bugfix] Fix failing transformers dynamic module resolving with spawn multiproc method (#13403)

Signed-off-by: Isotr0py <2037008807@qq.com>

hmellor committed:
[Doc]: Improve feature tables (#13224)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

Isotr0py committed:
[Bugfix] Remove noisy error logging during local model loading (#13458)

DarkLight1337 committed:
[ROCm] Make amdsmi import optional for other platforms (#13460)

zifeitong committed:
[Bugfix] Handle content type with optional parameters (#13383)

Signed-off-by: Zifei Tong <zifeitong@gmail.com>

liangfu committed:
[Bugfix] Fix invalid rotary embedding unit test (#13431)

Signed-off-by: Liangfu Chen <liangfc@amazon.com>

dtrifiro committed:
[CI/Build] migrate static project metadata from setup.py to pyproject.toml (#8772)

WoosukKwon committed:
[V1][PP] Enable true PP with Ray executor  (#13472)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

youkaichao committed:
[misc] fix debugging code (#13487)

Signed-off-by: youkaichao <youkaichao@gmail.com>

andoorve committed:
[V1][Tests] Adding additional testing for multimodal models to V1 (#13308)

Signed-off-by: andoorve <37849411+andoorve@users.noreply.github.com>

njhill committed:
[V1] Optimize handling of sampling metadata and req_ids list (#13244)

Signed-off-by: Nick Hill <nhill@redhat.com>

WoosukKwon committed:
Pin Ray version to 2.40.0 (#13490)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

WoosukKwon committed:
[V1][Spec Decode] Optimize N-gram matching with Numba (#13365)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

hmellor committed:
[Misc] Remove dangling references to `--use-v2-block-manager` (#13492)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

Commit:
[Hardware][Gaudi][Feature] Support Contiguous Cache Fetch  (#12139)

Signed-off-by: yuzhou <yuzhou@habana.ai>
Signed-off-by: zhouyu5 <yu.zhou@intel.com>
Co-authored-by: Cody Yu <hao.yu.cody@gmail.com>

khluu committed:
[perf-benchmark] Allow premerge ECR (#13509)

Signed-off-by: <>
Co-authored-by: EC2 Default User <ec2-user@ip-172-31-20-117.us-west-2.compute.internal>

divakar-amd committed:
[ROCm][MoE configs] mi325 mixtral & mi300 qwen_moe  (#13503)

ywang96 committed:
[Doc] Add clarification note regarding paligemma (#13511)

khluu committed:
[1/n][CI] Load models in CI from S3 instead of HF (#13205)

Signed-off-by: <>
Co-authored-by: EC2 Default User <ec2-user@ip-172-31-20-117.us-west-2.compute.internal>

khluu committed:
[perf-benchmark] Fix ECR path for premerge benchmark (#13512)

Signed-off-by: <>
Co-authored-by: EC2 Default User <ec2-user@ip-172-31-20-117.us-west-2.compute.internal>

Zzhiter committed:
use device param in load_model method (#13037)

alex-jw-brooks committed:
[Bugfix] Fix Positive Feature Layers in Llava Models (#13514)

Signed-off-by: Alex-Brooks <Alex.brooks@ibm.com>

luccafong committed:
[Model][Speculative Decoding] DeepSeek MTP spec decode (#12755)

Signed-off-by: Lu Fang <fanglu@fb.com>
Co-authored-by: LiuXiaoxuanPKU <lilyliupku@gmail.com>

njhill committed:
[V1][Core] Generic mechanism for handling engine utility (#13060)

Signed-off-by: Nick Hill <nhill@redhat.com>

yannicks1 committed:
[Feature] Pluggable platform-specific scheduler (#13161)

Signed-off-by: Yannick Schnider <yannick.schnider1@ibm.com>
Signed-off-by: Yannick Schnider <Yannick.Schnider1@ibm.com>

dtrifiro committed:
[CI/Build] force writing version file (#13544)

Signed-off-by: Daniele Trifirò <dtrifiro@redhat.com>

youkaichao committed:
[doc] clarify profiling is only for developers (#13554)

Signed-off-by: youkaichao <youkaichao@gmail.com>

DarkLight1337 committed:
[VLM][Bugfix] Pass processor kwargs properly on init (#13516)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>

ShangmingCai committed:
[Bugfix] Fix device ordinal for multi-node spec decode (#13269)

Signed-off-by: Shangming Cai <caishangming@linux.alibaba.com>

youkaichao committed:
[doc] clarify multi-node serving doc (#13558)

Signed-off-by: youkaichao <youkaichao@gmail.com>

wilsonwu committed:
Fix copyright year to auto get current year (#13561)

comaniac committed:
[MISC] Logging the message about Ray teardown (#13502)

Signed-off-by: Cody Yu <hao.yu.cody@gmail.com>
Co-authored-by: Rui Qiao <161574667+ruisearch42@users.noreply.github.com>

Isotr0py committed:
[Misc] Avoid calling unnecessary `hf_list_repo_files` for local model path (#13348)

Signed-off-by: isotr0py <2037008807@qq.com>

njhill committed:
[BugFix] Avoid error traceback in logs when V1 `LLM` terminates (#13565)

Signed-off-by: Nick Hill <nhill@redhat.com>

khluu committed:
[3/n][CI] Load Quantization test models with S3 (#13570)

Signed-off-by: <>
Co-authored-by: EC2 Default User <ec2-user@ip-172-31-20-117.us-west-2.compute.internal>

jeejeelee committed:
[Misc] Qwen2.5 VL support LoRA (#13261)

khluu committed:
[ci] Add AWS creds for AMD (#13572)

divakar-amd committed:
[ROCm][MoE] mi300 mixtral8x7B perf for specific BS (#13577)

Signed-off-by: Divakar Verma <divakar.verma@amd.com>

youkaichao committed:
[core] add sleep and wake up endpoint and v1 support (#12987)

Signed-off-by: youkaichao <youkaichao@gmail.com>
Signed-off-by: cennn <2523403608@qq.com>
Co-authored-by: cennn <2523403608@qq.com>

simon-mo committed:
[bugfix] spec decode worker get tp group only when initialized (#13578)

alex-jw-brooks committed:
[Misc] Warn if the vLLM version can't be retrieved (#13501)

Signed-off-by: Alex-Brooks <Alex.brooks@ibm.com>

wulipc committed:
[Misc] add mm_processor_kwargs to extra_body for Qwen2.5-VL (#13533)

gshtras committed:
[ROCm] MI300A compile targets deprecation (#13560)

terrytangyuan committed:
[API Server] Add port number range validation (#13506)

Signed-off-by: Yuan Tang <terrytangyuan@gmail.com>

mgoin committed:
[CI/Build] Use uv in the Dockerfile (#13566)

khluu committed:
[ci] Fix spec decode test (#13600)

khluu committed:
[2/n][ci] S3: Use full model path (#13564)

Signed-off-by: <>

varun-sundar-rabindranath committed:
[Kernel] LoRA - Refactor sgmv kernels (#13110)

hmellor committed:
Merge similar examples in `offline_inference` into single `basic` example (#12737)

chenxb002 committed:
[Bugfix] Fix deepseekv3 grouped topk error (#13474)

Signed-off-by: Chen-XiaoBing <chenxb002@whu.edu.cn>

hmellor committed:
Update `pre-commit`'s `isort` version to remove warnings (#13614)

WoosukKwon committed:
[V1][Minor] Print KV cache size in token counts (#13596)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

ajayvohra2005 committed:
fix neuron performance issue (#13589)

joerunde committed:
[Frontend] Add backend-specific options for guided decoding (#13505)

Signed-off-by: Joe Runde <Joseph.Runde@ibm.com>

mgoin committed:
[Bugfix] Fix max_num_batched_tokens for MLA (#13620)

Signed-off-by: mgoin <mgoin64@gmail.com>

lingfanyu committed:
[Neuron][Kernel] Vectorize KV cache load in FlashPagedAttention to maximize DMA bandwidth (#13245)

Signed-off-by: Lingfan Yu <lingfany@amazon.com>

kerthcet committed:
Add llmaz as another integration (#13643)

Signed-off-by: kerthcet <kerthcet@gmail.com>

Edwinhr716 committed:
[Misc] Adding script to setup ray for multi-node vllm deployments  (#12913)

kaixih committed:
[NVIDIA] Fix an issue to use current stream for the nvfp4 quant (#13632)

hmellor committed:
Use pre-commit to update `requirements-test.txt` (#13617)

ywang96 committed:
[Bugfix] Add `mm_processor_kwargs` to chat-related protocols (#13644)

njhill committed:
[V1][Sampler] Avoid an operation during temperature application (#13587)

SzymonOzog committed:
Missing comment explaining VDR variable in GGUF kernels (#13290)

gmarinho2 committed:
[FEATURE] Enables /score endpoint for embedding models (#12846)

khluu committed:
[ci] Fix metrics test model path (#13635)

Hongbosherlock committed:
[Kernel]Add streamK for block-quantized CUTLASS kernels (#12978)

Isotr0py committed:
[Bugfix][CPU] Fix cpu all-reduce using native pytorch implementation  (#13586)

johnzheng1975 committed:
fix typo of grafana dashboard, with correct datasource (#13668)

Signed-off-by: John Zheng <john.zheng@hp.com>

LucasWilkinson committed:
[Attention] MLA with chunked prefill (#12639)

Signed-off-by: Lucas Wilkinson <lwilkinson@neuralmagic.com>
Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>
Co-authored-by: Patrick Horn <patrick.horn@gmail.com>
Co-authored-by: simon-mo <xmo@berkeley.edu>
Co-authored-by: Tyler Michael Smith <tyler@neuralmagic.com>

Isotr0py committed:
[Misc] Fix yapf linting tools etc not running on pre-commit (#13695)

Signed-off-by: Isotr0py <2037008807@qq.com>

terrytangyuan committed:
docs: Add a note on full CI run in contributing guide (#13646)

youngkent committed:
[HTTP Server] Make model param optional in request (#13568)

WangErXiao committed:
[Bugfix][API Server] Fix invalid usage of 'ge' and 'le' in port valid… (#13672)

waltforme committed:
[Misc] Capture and log the time of loading weights (#13666)

gongdao123 committed:
[ROCM] fix native attention function call (#13650)

2015aroras committed:
[Bugfix][Model] OLMo 2: split qkv correctly for GQA and MQA (#13687)

dsikka committed:
[Misc] Bump compressed-tensors (#13619)

WangErXiao committed:
[Bugfix] Fix benchmark script bug: inaccurate stats for vllm backend when max_model_len < input_len + output_len (#13691)

Signed-off-by: WangErXiao <863579016@qq.com>

houseroad committed:
[v1] Support allowed_token_ids in v1 Sampler (#13210)

Signed-off-by: Lu Fang <lufang@fb.com>

JenZhao committed:
[Bugfix] V1 Memory Profiling: V0 Sampler Integration without Rejection Sampler (#13594)

Signed-off-by: Jennifer Zhao <7443418+JenZhao@users.noreply.github.com>
Co-authored-by: Roger Wang <ywang@roblox.com>

fabianlim committed:
Correction to TP logic for Mamba Mixer 2 when Num Groups not divisible by TP Size (#13660)

markmc committed:
[V1][Metrics] Support `vllm:cache_config_info` (#13299)

markmc committed:
[Metrics] Add `--show-hidden-metrics-for-version` CLI arg (#13295)

jeejeelee committed:
[Misc] Reduce LoRA-related static variable (#13166)

DarkLight1337 committed:
[CI/Build] Fix pre-commit errors (#13696)

youkaichao committed:
[core] set up data parallel communication (#13591)

Signed-off-by: youkaichao <youkaichao@gmail.com>

youkaichao committed:
[ci] fix linter (#13701)

Signed-off-by: youkaichao <youkaichao@gmail.com>

youngkent committed:
Support SSL Key Rotation in HTTP Server (#13495)

kaixih committed:
[NVIDIA] Support nvfp4 cutlass gemm (#13571)

SageMoore committed:
[V1][Kernel] Refactor the prefix_prefill kernel so that the caller no longer has to pass in the context lengths (#13095)

gshtras committed:
[ROCm] Apply FP8 weights padding to values not divisible by 512 bytes on ROCm (#13231)

DarkLight1337 committed:
[Doc] Dockerfile instructions for optional dependencies and dev transformers (#13699)

helena-intel committed:
[Bugfix] Fix boolean conversion for OpenVINO env variable (#13615)

yma11 committed:
[XPU]fix setuptools version for xpu (#13548)

dtrifiro committed:
[CI/Build] fix uv caching in Dockerfile (#13611)

ywang96 committed:
[CI/Build] Fix pre-commit errors from #13571 (#13709)

Signed-off-by: Roger Wang <ywang@roblox.com>

andylolu2 committed:
[BugFix] Minor: logger import in attention backend (#13706)

Signed-off-by: Andy Lo <andy@mistral.ai>

khluu committed:
[ci] Use env var to control whether to use S3 bucket in CI (#13634)

kylesayrs committed:
[Quant] BaiChuan SupportsQuant (#13710)

Isotr0py committed:
[LMM] Implement merged multimodal processor for whisper (#13278)

njhill committed:
[Core][Distributed] Use IPC (domain socket) ZMQ socket for local comms (#13688)

ywang96 committed:
[Misc] Deprecate `--dataset` from `benchmark_serving.py` (#13708)

Signed-off-by: Roger Wang <ywang@roblox.com>

youkaichao committed:
[v1] torchrun compatibility (#13642)

Signed-off-by: youkaichao <youkaichao@gmail.com>

njhill committed:
[V1][BugFix] Fix engine core client shutdown hangs (#13298)

Even though ZMQ context.destroy() is meant to close open sockets before terminating the context, it appears to be necessary to do this explicitly or else it can hang in the context.term() method.

Close zmq sockets explicitly before terminating context, make shutdown of client resource more robust, shut down engine core process prior to terminating zmq context.

Signed-off-by: Nick Hill <nhill@redhat.com>

huydhn committed:
Fix some issues with benchmark data output (#13641)

Signed-off-by: Huy Do <huydhn@gmail.com>

khluu committed:
[ci] Add logic to change model to S3 path only when S3 CI env var is on (#13727)

Signed-off-by: <>
Co-authored-by: EC2 Default User <ec2-user@ip-172-31-63-253.us-west-2.compute.internal>

ywang96 committed:
[V1][Core] Fix memory issue with logits & sampling (#13721)

MengqingCao committed:
[model][refactor] remove cuda hard code in models and layers (#13658)

bufferoverflow committed:
[Bugfix] fix(logging): add missing opening square bracket (#13011)

bufferoverflow committed:
[CI/Build] add python-json-logger to requirements-common (#12842)

Akshat-Tripathi committed:
Fixed model compilation bugs

Signed-off-by: Akshat Tripathi <akshat@krai.ai>

cakeng committed:
Expert Parallelism (EP) Support for DeepSeek V2 (#12583)

Abatom committed:
[BugFix]  Illegal memory access for MoE On H20 (#13693)

NickLucche committed:
[Misc][Docs] Raise error when flashinfer is not installed and `VLLM_ATTENTION_BACKEND` is set (#12513)

Signed-off-by: NickLucche <nlucches@redhat.com>

afeldman-nm committed:
[V1] V1 engine implements parallel sampling (AsyncLLM and LLMEngine) (#10980)

Signed-off-by: Andrew Feldman <afeldman@neuralmagic.com>
Co-authored-by: Nick Hill <nhill@redhat.com>

ywang96 committed:
Revert "[V1][Core] Fix memory issue with logits & sampling" (#13775)

mgoin committed:
Fix precommit fail in fused_moe intermediate_cache2 chunking (#13772)

Signed-off-by: mgoin <mgoin64@gmail.com>

robertgshaw2-redhat committed:
[Misc] Clean Up `EngineArgs.create_engine_config` (#13734)

Signed-off-by: rshaw@neuralmagic.com <rshaw@neuralmagic.com>

robertgshaw2-redhat committed:
[Misc][Chore] Clean Up `AsyncOutputProcessing` Logs (#13780)

hmellor committed:
Remove unused kwargs from model definitions (#13555)

eli-b committed:
[Doc] arg_utils.py: fixed a typo (#13785)

cjackal committed:
[Misc] set single whitespace between log sentences (#13771)

Signed-off-by: cjackal <44624812+cjackal@users.noreply.github.com>

tlrmchlsmth committed:
[Bugfix][Quantization] Fix FP8 + EP (#13784)

Signed-off-by: Tyler Michael Smith <tyler@neuralmagic.com>

wangxiyuan committed:
[Misc][Attention][Quantization] init property earlier (#13733)

Signed-off-by: wangxiyuan <wangxiyuan1007@gmail.com>

markmc committed:
[V1][Metrics] Implement vllm:lora_requests_info metric (#13504)

LucasWilkinson committed:
[Bugfix] Fix deepseek-v2 error: "missing 1 required positional argument: 'residual'" (#13802)

mgoin committed:
[Bugfix] Support MLA for CompressedTensorsWNA16 (#13725)

Signed-off-by: mgoin <mgoin64@gmail.com>

mgoin committed:
Fix CompressedTensorsWNA16MoE with grouped scales (#13769)

varun-sundar-rabindranath committed:
[Core] LoRA V1 - Add add/pin/list/remove_lora functions   (#13705)

DarkLight1337 committed:
[Misc] Check that the model can be inspected upon registration (#13743)

russellb committed:
[Core] xgrammar: Expand list of unsupported jsonschema keywords (#13783)

Signed-off-by: Russell Bryant <rbryant@redhat.com>

shen-shanshan committed:
[Bugfix] Modify modelscope api usage in transformer_utils (#13807)

ruisearch42 committed:
[misc] Clean up ray compiled graph type hints (#13731)

YaoJiayi committed:
[Feature] Support KV cache offloading and disagg prefill with LMCache connector. (#12953)

gshtras committed:
[ROCm][Quantization][Kernel] Using HIP FP8 header (#12593)

jeejeelee committed:
[CI/Build]  Fix V1 LoRA failure (#13767)

Chen-0210 committed:
[Misc]Clarify Error Handling for Non-existent Model Paths and HF Repo IDs (#13724)

Signed-off-by: Chen-0210 <chenjincong11@gmail.com>
Co-authored-by: Michael Goin <mgoin64@gmail.com>

edwardzjl committed:
[Bugfix] Initialize attention bias on the same device as Query/Key/Value (#13468)

naromero77amd committed:
[Bugfix] Flush TunableOp results before worker processes are destroyed. (#13623)

Signed-off-by: Nichols A. Romero <nick.romero@amd.com>

Isotr0py committed:
[Bugfix] Fix deepseek-vl2 inference with more than 2 images (#13818)

HermitSun committed:
Fix `/v1/audio/transcriptions ` Bad Request Error (#13811)

DarkLight1337 committed:
[Bugfix] Revert inspection code in #13743 (#13832)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>

Akshat-Tripathi committed:
Minor changes

Signed-off-by: Akshat Tripathi <akshat@krai.ai>

Chen-0210 committed:
Fix string parsing error (#13825)

liangfu committed:
[Neuron] Add custom_ops for neuron backend (#13246)

Signed-off-by: Liangfu Chen <liangfc@amazon.com>
Co-authored-by: George Novack <gnovack@amazon.com>
Co-authored-by: Aoyu Zhang <aoyuzhan@amazon.com>

hmellor committed:
Fix failing `MyGemma2Embedding` test (#13820)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

mgoin committed:
[Model] Support Grok1 (#13795)

Signed-off-by: mgoin <mgoin64@gmail.com>

hmellor committed:
DeepSeek V2/V3/R1 only place `lm_head` on last pp rank (#13833)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

ruisearch42 committed:
[misc] Show driver IP info when Ray fails to allocate driver worker (#13858)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>

LiuXiaoxuanPKU committed:
[V1][Spec Decode] Change Spec Decode Rejection Sampling API (#13729)

noemotiovon committed:
[Misc]Code Cleanup (#13859)

Signed-off-by: noemotiovon <noemotiovon@gmail.com>
Co-authored-by: noemotiovon <noemotiovon@gmail.com>

henrylhtsang committed:
[Kernel][Build/CI] Bump CUTLASS to 3.8 and add initializers for cutlass epilogues (#13797)

hmellor committed:
Improve pipeline partitioning (#13839)

tianyuzhou95 committed:
[Doc] fix the incorrect module path of tensorize_vllm_model (#13863)

SageMoore committed:
[ROCm] Disable chunked prefill/prefix caching when running MLA on non-cuda platforms (#13844)

Signed-off-by: Sage Moore <sage@neuralmagic.com>

sethkimmel3 committed:
[v0][Core] Use xgrammar shared context to avoid copy overhead for offline engine (#13837)

Signed-off-by: Seth Kimmel <seth.kimmel3@gmail.com>

ywang96 committed:
[Misc] Fix input processing for Ultravox (#13871)

DarkLight1337 committed:
[Bugfix] Add test example for Ultravox v0.5 (#13890)

hmellor committed:
Add comments on accessing `kv_cache` and `attn_metadata` (#13887)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

fgreinacher committed:
[Bugfix] Handle None parameters in Mistral function calls. (#13786)

b8zhong committed:
[Misc]: Add support for goodput on guided benchmarking + TPOT calculation refactor (#13736)

Signed-off-by: Brayden Zhong <b8zhong@uwaterloo.ca>

joerunde committed:
[Bugfix] Do not crash V0 engine on input errors (#13101)

Signed-off-by: Joe Runde <Joseph.Runde@ibm.com>

DarkLight1337 committed:
[Bugfix] Update expected token counts for Ultravox tests (#13895)

Chenyaaang committed:
[TPU] use torch2.6 with whl package (#13860)

Signed-off-by: Chenyaaang <llccyy1212@gmail.com>

chaunceyjiang committed:
[Misc] fixed qwen_vl_utils parameter error (#13906)

wallashss committed:
[Bugfix] Backend option to disable xgrammar any_whitespace (#12744)

Signed-off-by: Wallas Santos <wallashss@ibm.com>
Signed-off-by: Joe Runde <Joseph.Runde@ibm.com>
Co-authored-by: Joe Runde <Joseph.Runde@ibm.com>

WoosukKwon committed:
[BugFix] Make FP8 Linear compatible with torch.compile (#13918)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

LucasWilkinson committed:
[Kernel] FlashMLA integration (#13747)

Signed-off-by: Lucas Wilkinson <lwilkinson@neuralmagic.com>
Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

HollowMan6 committed:
[ROCm][Quantization][Kernel] Use FP8 FNUZ when OCP flag is 0 or undefined (#13851)

Signed-off-by: Hollow Man <hollowman@opensuse.org>

mgoin committed:
Use CUDA 12.4 as default for release and nightly wheels (#12098)

ruisearch42 committed:
[misc] Rename Ray ADAG to Compiled Graph (#13928)

SageMoore committed:
[ROCm][V1] Update reshape_and_cache to properly work with CUDA graph padding (#13922)

markmc committed:
[V1][Metrics] Handle preemptions (#13169)

b8zhong committed:
[CI/Build] Add examples/ directory to be labelled by `mergify` (#13944)

Signed-off-by: Brayden Zhong <b8zhong@uwaterloo.ca>

chaunceyjiang committed:
[Misc] fixed 'required' is an invalid argument for positionals (#13948)

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>

zhengy001 committed:
[PP] Correct cache size check (#13873)

Signed-off-by: Yang Zheng <zhengy.gator@gmail.com>

mgoin committed:
Fix test_block_fp8.py test for MoE (#13915)

Signed-off-by: mgoin <mgoin64@gmail.com>

Isotr0py committed:
[VLM] Support multimodal inputs for Florence-2 models (#13320)

SzymonOzog committed:
[Model] Deepseek GGUF support  (#13167)

observerw committed:
Update quickstart.md (#13958)

hmellor committed:
Deduplicate `.pre-commit-config.yaml`'s `exclude` (#13967)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

ruisearch42 committed:
[bugfix] Fix profiling for RayDistributedExecutor (#13945)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>

noamgat committed:
Update LMFE version to v0.10.11 to support new versions of transforme… (#13930)

Isotr0py committed:
[Bugfix] Fix qwen2.5-vl overflow issue (#13968)

Signed-off-by: Isotr0py <2037008807@qq.com>

DarkLight1337 committed:
[VLM] Generalized prompt updates for multi-modal processor (#13964)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>

Commit:
[Attention] MLA support for V1 (#13789)

Signed-off-by: Yang Chen <yangche@fb.com>

dependabot[bot] committed:
Bump azure/setup-helm from 4.2.0 to 4.3.0 (#13742)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

DarkLight1337 committed:
[VLM] Deprecate legacy input mapper for OOT multimodal models (#13979)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>

SageMoore committed:
[ROCm] Fix the Kernels, Core, and Prefix Caching AMD CI groups (#13970)

Signed-off-by: Sage Moore <sage@neuralmagic.com>

WoosukKwon committed:
[V1][Minor] Minor cleanup for GPU Model Runner (#13983)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

qli88 committed:
[core] Perf improvement for DSv3 on AMD GPUs (#13718)

Signed-off-by: qli88 <qiang.li2@amd.com>

LucasWilkinson committed:
[Attention] Flash MLA for V1 (#13867)

Signed-off-by: Yang Chen <yangche@fb.com>
Signed-off-by: Lucas Wilkinson <lwilkinson@neuralmagic.com>
Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>
Co-authored-by: Yang Chen <yangche@fb.com>

benchislett committed:
[Model][Speculative Decoding] Expand DeepSeek MTP code to support k > n_predict (#13626)

Signed-off-by: Benjamin Chislett <benjamin.chislett@centml.ai>

jeejeelee committed:
[Misc] Print FusedMoE detail info (#13974)

ywang96 committed:
[V1]`SupportsV0Only` protocol for model definitions (#13959)

Signed-off-by: Roger Wang <ywang@roblox.com>

tjohnson31415 committed:
[Bugfix] Check that number of images matches number of <|image|> tokens with mllama (#13911)

Signed-off-by: Travis Johnson <tsjohnso@us.ibm.com>

DarkLight1337 committed:
[Doc] Move multimodal Embedding API example to Online Serving page (#14017)

Signed-off-by: DarkLight1337 <tlleungac@connect.ust.hk>

hasB4K committed:
[Bugfix][Disaggregated] patch the inflight batching on the decode node in SimpleConnector to avoid hangs in SimpleBuffer (nccl based) (#13987)

Signed-off-by: Mathis Felardos <mathis@mistral.ai>

hmellor committed:
Use smaller embedding model when not testing model specifically (#13891)

Kacper-Pietkun committed:
[Hardware][Intel-Gaudi] Regional compilation support (#13213)

Ryp committed:
[V1][Minor] Restore V1 compatibility with LLMEngine class (#13090)

hmellor committed:
Update AutoAWQ docs (#14042)

Signed-off-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>

jeejeelee committed:
[Bugfix] Fix MoeWNA16Method activation (#14024)

DarkLight1337 committed:
[VLM][Bugfix] Enable specifying prompt target via index (#14038)

LouieYang committed:
[Bugfix] Initialize attention bias on the same device as Query/Key/Value for QwenVL Series (#14031)

b8zhong committed:
[Doc] Fix ROCm documentation (#14041)

Signed-off-by: Brayden Zhong <b8zhong@uwaterloo.ca>

hmellor committed:
Fix entrypoint tests for embedding models (#14052)

heheda12345 committed:
[v1] Cleanup the BlockTable in InputBatch (#13977)

Signed-off-by: Chen Zhang <zhangch99@outlook.com>

atalman committed:
Add RELEASE.md (#13926)

Signed-off-by: atalman <atalman@fb.com>

heheda12345 committed:
[v1] Move block pool operations to a separate class (#13973)

Signed-off-by: Chen Zhang <zhangch99@outlook.com>
Co-authored-by: Cody Yu <hao.yu.cody@gmail.com>

ruisearch42 committed:
[core] Bump ray to 2.43 (#13994)

Signed-off-by: Rui Qiao <ruisearch42@gmail.com>

ProExpertProg committed:
[torch.compile] Fix RMSNorm + quant fusion in the non-cutlass-fp8 case, rename RedundantReshapesPass to NoopEliminationPass (#10902)

Signed-off-by: luka <luka@neuralmagic.com>

b8zhong committed:
[Docs] Add `pipeline_parallel_size` to optimization docs (#14059)

Signed-off-by: Brayden Zhong <b8zhong@uwaterloo.ca>

jeejeelee committed:
[Bugfix] Add file lock for ModelScope download (#14060)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>

wyajieha committed:
[Misc][Kernel]: Add GPTQAllSpark Quantization (#12931)

bigPYJ1151 committed:
[Bugfix][V1][Minor] Fix shutting_down flag checking in V1 MultiprocExecutor (#14053)

KuntaiDu committed:
[Documentation] Add more deployment guide for Kubernetes deployment (#13841)

Signed-off-by: KuntaiDu <kuntai@uchicago.edu>
Signed-off-by: Kuntai Du <kuntai@uchicago.edu>

Isotr0py committed:
[Doc] Consolidate `whisper` and `florence2` examples (#14050)

WoosukKwon committed:
[V1][Minor] Do not print attn backend twice (#13985)

Signed-off-by: Woosuk Kwon <woosuk.kwon@berkeley.edu>

SageMoore committed:
[ROCm][V1][Bugfix] Add get_builder_cls method to the ROCmAttentionBackend class (#14065)

Signed-off-by: Sage Moore <sage@neuralmagic.com>

heheda12345 committed:
[v1][Bugfix] Only cache blocks that are not in the prefix cache (#14073)

heheda12345 committed:
[v1] Add `__repr__` to KVCacheBlock to avoid recursive print (#14081)

jeejeelee committed:
[Model] Add LoRA support for TransformersModel (#13770)

Signed-off-by: Jee Jee Li <pandaleefree@gmail.com>

waltforme committed:
[Misc] Accurately capture the time of loading weights (#14063)

Signed-off-by: Jun Duan <jun.duan.phd@outlook.com>

qux-bbb committed:
[Doc] Source building add clone step (#14086)

Signed-off-by: qux-bbb <1147635419@qq.com>

gaocegege committed:
[v0][structured output] Support reasoning output (#12955)

Signed-off-by: Ce Gao <cegao@tensorchord.ai>

hmellor committed:
Update deprecated Python 3.8 typing (#13971)

realShengYao committed:
[Bugfix] Explicitly include "omp.h" for MacOS to avoid installation failure (#14051)

noooop committed:
[Misc] duplicate code in deepseek_v2  (#14106)

MengqingCao committed:
[Misc][Platform] Move use allgather to platform (#14010)

Signed-off-by: Mengqing Cao <cmq0113@163.com>

comaniac committed:
[Build] Make sure local main branch is synced when VLLM_USE_PRECOMPILED=1 (#13921)

Signed-off-by: Cody Yu <hao.yu.cody@gmail.com>

markmc committed:
[V1] Refactor parallel sampling support (#13774)

Signed-off-by: Mark McLoughlin <markmc@redhat.com>

Akshat-Tripathi committed:
[Misc] Improve LoRA spelling (#13831)

Akshat-Tripathi committed:
[V1][TPU] Integrate the new ragged paged attention kernel with vLLM v1 on TPU (#13379)

Signed-off-by: Xiongfei Wei <isaacwxf23@gmail.com>
Signed-off-by: mgoin <mgoin64@gmail.com>
Co-authored-by: mgoin <mgoin64@gmail.com>

Akshat-Tripathi commented:
Closing in favour of https://github.com/vllm-project/vllm/pull/14238
